research/conditional/train/cc_train.py

```diff
@@ -92,8 +91,8 @@ def main(
         torch.autograd.set_detect_anomaly(True)
 
     data_distributed = True if rank is not None else False
-    ff_layer_fun = get_ff_layer(args)
-    attention_layer_fun = get_attention_layer(args)
+    ff_layer_fun = get_ff_layer(args, rank)
+    attention_layer_fun = get_attention_layer(args, rank=rank, fsdp_enabled=args.fsdp)
```

Comment:
how is fsdp enabled relevant only in attention?

Reply:
you are right, `get_attention_layer` should take this info from `args`